alibi/explainers/similarity/grad.py

```diff
@@ -5,82 +5,139 @@
 
 import copy
 from typing import TYPE_CHECKING, Callable, Optional, Union, Dict, Tuple
+from typing_extensions import Literal
+
+from enum import Enum
 
 import numpy as np
 
 from alibi.api.interfaces import Explanation
 from alibi.explainers.similarity.base import BaseSimilarityExplainer
 from alibi.explainers.similarity.metrics import dot, cos, asym_dot
 from alibi.api.defaults import DEFAULT_META_SIM, DEFAULT_DATA_SIM
+from alibi.utils.frameworks import Framework
+from alibi.api.interfaces import Explainer
 
 if TYPE_CHECKING:
     import tensorflow
     import torch
 
 
-class SimilarityExplainer(BaseSimilarityExplainer):
+class Task(Enum):
+    """
+    Enum of supported tasks.
+    """
+    CLASSIFICATION = "classification"
+    REGRESSION = "regression"
+
+    @staticmethod
+    def from_str(name: str):
+        return {
+            'classification': Task.CLASSIFICATION,
+            'regression': Task.REGRESSION
+        }[name]
+
+    @staticmethod
+    def values():
+        return {item.value for item in Task.__members__.values()}
+
+    @staticmethod
+    def options_string():
+        return f"""'{"' | '".join(Task.values())}'"""
+
+
+class GradientSimilarity(BaseSimilarityExplainer):
+
     def __init__(self,
                  predictor: 'Union[tensorflow.keras.Model, torch.nn.Module]',
-                 loss_fn: '''Callable[[Union[tensorflow.Tensor, torch.Tensor],
-                                    Union[tensorflow.Tensor, torch.Tensor]],
-                                   Union[tensorflow.Tensor, torch.Tensor]]''',
-                 sim_fn: 'Union[Callable[[np.ndarray, np.ndarray], np.ndarray], str]' = 'grad_dot',
-                 task: str = "classification",
+                 loss_fn: '''Union[Callable[[tensorflow.Tensor, tensorflow.Tensor], tensorflow.Tensor],
+                                   Callable[[torch.Tensor, torch.Tensor], torch.Tensor]]''',
+                 sim_fn: Literal['grad_dot', 'grad_cos', 'grad_asym_dot'] = 'grad_dot',
```

Comment:
Did we decide to not support custom similarities (for now)?

Reply:
I don't actually recall ever discussing the possibility? Shall I add that functionality or defer it for now?

Reply:
No, let's leave it narrowly scoped for now. I just thought we might have because the previous type-hints had `Callable`.